### PR TITLE
Renove bl

### PR DESCRIPTION
**Pull request by benallfree**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).

- [x] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## readable-stream ([<kbd>unpkg</kbd>](https://unpkg.com/browse/readable-stream@latest/))
was missing the following properties:
1. from
1. finished
1. pipeline
1. once
1. on

<details>
<summary>as well as these 6 other properties...</summary>
<p>listenerCount, errorMonitor, captureRejectionSymbol, captureRejections, defaultMaxListeners, EventEmitter</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against aee8767ac91fc5a226dd8b036361b2b53366f7bd
</p>

File: types/readable-stream/index.d.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@
 // Project: https://github.com/nodejs/readable-stream
 // Definitions by: TeamworkGuy2 <https://github.com/TeamworkGuy2>
 //                   markdreyer <https://github.com/markdreyer>
+//                   benallfree <https://github.com/benallfree>
```
Comment (jakebailey) [inline types/readable-stream/index.d.ts:5]: Are you intending to become an owner of these packages? You don't need to put your username here if you are juts looking for contribution credit.

File: types/readable-stream/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
     "private": true,
     "dependencies": {
-        "safe-buffer": "*"
+        "safe-buffer": "^5.1.2"
```
Comment (jakebailey) [inline types/readable-stream/package.json:4]: Is this change related? This doesn't mention bl...

Comment (benallfree) [reply types/readable-stream/package.json:4]: Yeah there is some kind of transient dependency issue happening. See previous CI failure https://github.com/DefinitelyTyped/DefinitelyTyped/commit/4753fab991f3a87c85b883f4c23193732e15f9b4/checks?check_suite_id=8944884489#step:6:49

I actually get a different error locally, the one about unnamed external dependencies. 

You might be able to help with this. `msgpack5` imports `bl` which imports `readable-stream`, which imports `safe-buffer` https://github.com/DefinitelyTyped/DefinitelyTyped/blob/8d76592985f2965200c124036f9cc458fb4ed97c/types/readable-stream/index.d.ts#L10.

`*` was causing CI failures I think due to pulling in a very old version into `node_modules`. By pointing to a specific version >5.1.2 that includes typings, the CI error cleared up. 

Here's the new `npm why safe-buffer`, which shows that the old `2.1` version CI was complaining about is gone:

```bash
npm why safe-buffer
safe-buffer@5.2.1 dev
node_modules/safe-buffer
  safe-buffer@"^5.0.1" from ecdsa-sig-formatter@1.0.11
  node_modules/ecdsa-sig-formatter
    ecdsa-sig-formatter@"1.0.11" from jwa@1.4.1
    node_modules/jwa
      jwa@"^1.4.1" from jws@3.2.2
      node_modules/jws
        jws@"^3.2.2" from jsonwebtoken@8.5.1
        node_modules/jsonwebtoken
          jsonwebtoken@"^8.4.0" from danger@10.9.0
          node_modules/danger
            dev danger@"^10.1.1" from the root project
  safe-buffer@"^5.0.1" from jwa@1.4.1
  node_modules/jwa
    jwa@"^1.4.1" from jws@3.2.2
    node_modules/jws
      jws@"^3.2.2" from jsonwebtoken@8.5.1
      node_modules/jsonwebtoken
        jsonwebtoken@"^8.4.0" from danger@10.9.0
        node_modules/danger
          dev danger@"^10.1.1" from the root project
  safe-buffer@"^5.0.1" from jws@3.2.2
  node_modules/jws
    jws@"^3.2.2" from jsonwebtoken@8.5.1
    node_modules/jsonwebtoken
      jsonwebtoken@"^8.4.0" from danger@10.9.0
      node_modules/danger
        dev danger@"^10.1.1" from the root project
  safe-buffer@"^5.1.2" from request@2.88.2
  node_modules/request
    request@"^2.88.2" from @qiwi/npm-registry-client@8.9.1
    node_modules/@qiwi/npm-registry-client
      @qiwi/npm-registry-client@"^8.9.1" from @definitelytyped/utils@0.0.134
      node_modules/@definitelytyped/utils
        dev @definitelytyped/utils@"latest" from the root project
        @definitelytyped/utils@"^0.0.134" from @definitelytyped/dtslint@0.0.134
        node_modules/@definitelytyped/dtslint
          dev @definitelytyped/dtslint@"latest" from the root project
          @definitelytyped/dtslint@"^0.0.134" from @definitelytyped/dtslint-runner@0.0.134
          node_modules/@definitelytyped/dtslint-runner
            dev @definitelytyped/dtslint-runner@"latest" from the root project
        @definitelytyped/utils@"^0.0.134" from @definitelytyped/definitions-parser@0.0.134
        node_modules/@definitelytyped/definitions-parser
          dev @definitelytyped/definitions-parser@"latest" from the root project
          @definitelytyped/definitions-parser@"^0.0.134" from @definitelytyped/dtslint-runner@0.0.134
          node_modules/@definitelytyped/dtslint-runner
            dev @definitelytyped/dtslint-runner@"latest" from the root project
        @definitelytyped/utils@"^0.0.134" from @definitelytyped/dtslint-runner@0.0.134
        node_modules/@definitelytyped/dtslint-runner
          dev @definitelytyped/dtslint-runner@"latest" from the root project
  safe-buffer@"~5.2.0" from string_decoder@1.3.0
  node_modules/string_decoder
    string_decoder@"^1.1.1" from readable-stream@3.6.0
    node_modules/readable-stream
      readable-stream@"^3.4.0" from bl@4.1.0
      node_modules/bl
        bl@"^4.0.3" from tar-stream@2.2.0
        node_modules/tar-stream
          tar-stream@"^2.1.4" from @definitelytyped/utils@0.0.134
          node_modules/@definitelytyped/utils
            dev @definitelytyped/utils@"latest" from the root project
            @definitelytyped/utils@"^0.0.134" from @definitelytyped/dtslint@0.0.134
            node_modules/@definitelytyped/dtslint
              dev @definitelytyped/dtslint@"latest" from the root project
              @definitelytyped/dtslint@"^0.0.134" from @definitelytyped/dtslint-runner@0.0.134
              node_modules/@definitelytyped/dtslint-runner
                dev @definitelytyped/dtslint-runner@"latest" from the root project
            @definitelytyped/utils@"^0.0.134" from @definitelytyped/definitions-parser@0.0.134
            node_modules/@definitelytyped/definitions-parser
              dev @definitelytyped/definitions-parser@"latest" from the root project
              @definitelytyped/definitions-parser@"^0.0.134" from @definitelytyped/dtslint-runner@0.0.134
              node_modules/@definitelytyped/dtslint-runner
                dev @definitelytyped/dtslint-runner@"latest" from the root project
            @definitelytyped/utils@"^0.0.134" from @definitelytyped/dtslint-runner@0.0.134
            node_modules/@definitelytyped/dtslint-runner
              dev @definitelytyped/dtslint-runner@"latest" from the root project
      readable-stream@"^3.0.2" from concat-stream@2.0.0
      node_modules/concat-stream
        concat-stream@"^2.0.0" from unified-engine@10.0.1
        node_modules/unified-engine
          unified-engine@"^10.0.1" from remark-validate-links@12.1.0
          node_modules/remark-validate-links
            dev remark-validate-links@"^12.0.0" from the root project
          unified-engine@"^10.0.0" from unified-args@10.0.0
          node_modules/unified-args
            unified-args@"^10.0.0" from remark-cli@11.0.0
            node_modules/remark-cli
              dev remark-cli@"^11.0.0" from the root project
        concat-stream@"^2.0.0" from @qiwi/npm-registry-client@8.9.1
        node_modules/@qiwi/npm-registry-client
          @qiwi/npm-registry-client@"^8.9.1" from @definitelytyped/utils@0.0.134
          node_modules/@definitelytyped/utils
            dev @definitelytyped/utils@"latest" from the root project
            @definitelytyped/utils@"^0.0.134" from @definitelytyped/dtslint@0.0.134
            node_modules/@definitelytyped/dtslint
              dev @definitelytyped/dtslint@"latest" from the root project
              @definitelytyped/dtslint@"^0.0.134" from @definitelytyped/dtslint-runner@0.0.134
              node_modules/@definitelytyped/dtslint-runner
                dev @definitelytyped/dtslint-runner@"latest" from the root project
            @definitelytyped/utils@"^0.0.134" from @definitelytyped/definitions-parser@0.0.134
            node_modules/@definitelytyped/definitions-parser
              dev @definitelytyped/definitions-parser@"latest" from the root project
              @definitelytyped/definitions-parser@"^0.0.134" from @definitelytyped/dtslint-runner@0.0.134
              node_modules/@definitelytyped/dtslint-runner
                dev @definitelytyped/dtslint-runner@"latest" from the root project
            @definitelytyped/utils@"^0.0.134" from @definitelytyped/dtslint-runner@0.0.134
            node_modules/@definitelytyped/dtslint-runner
              dev @definitelytyped/dtslint-runner@"latest" from the root project
      readable-stream@"^3.1.1" from tar-stream@2.2.0
      node_modules/tar-stream
        tar-stream@"^2.1.4" from @definitelytyped/utils@0.0.134
        node_modules/@definitelytyped/utils
          dev @definitelytyped/utils@"latest" from the root project
          @definitelytyped/utils@"^0.0.134" from @definitelytyped/dtslint@0.0.134
          node_modules/@definitelytyped/dtslint
            dev @definitelytyped/dtslint@"latest" from the root project
            @definitelytyped/dtslint@"^0.0.134" from @definitelytyped/dtslint-runner@0.0.134
            node_modules/@definitelytyped/dtslint-runner
              dev @definitelytyped/dtslint-runner@"latest" from the root project
          @definitelytyped/utils@"^0.0.134" from @definitelytyped/definitions-parser@0.0.134
          node_modules/@definitelytyped/definitions-parser
            dev @definitelytyped/definitions-parser@"latest" from the root project
            @definitelytyped/definitions-parser@"^0.0.134" from @definitelytyped/dtslint-runner@0.0.134
            node_modules/@definitelytyped/dtslint-runner
              dev @definitelytyped/dtslint-runner@"latest" from the root project
          @definitelytyped/utils@"^0.0.134" from @definitelytyped/dtslint-runner@0.0.134
          node_modules/@definitelytyped/dtslint-runner
            dev @definitelytyped/dtslint-runner@"latest" from the root project
      readable-stream@"^3.6.0" from are-we-there-yet@3.0.1
      node_modules/node-gyp/node_modules/are-we-there-yet
        are-we-there-yet@"^3.0.0" from npmlog@6.0.2
        node_modules/node-gyp/node_modules/npmlog
          npmlog@"^6.0.0" from node-gyp@9.3.0
          node_modules/node-gyp
            node-gyp@"^9.0.0" from @npmcli/run-script@4.2.1
            node_modules/@npmcli/run-script
              @npmcli/run-script@"^4.1.0" from pacote@13.6.2
              node_modules/pacote
                pacote@"^13.6.1" from @definitelytyped/definitions-parser@0.0.134
                node_modules/@definitelytyped/definitions-parser
                  dev @definitelytyped/definitions-parser@"latest" from the root project
                  @definitelytyped/definitions-parser@"^0.0.134" from @definitelytyped/dtslint-runner@0.0.134
                  node_modules/@definitelytyped/dtslint-runner
                    dev @definitelytyped/dtslint-runner@"latest" from the root project
                pacote@"^13.6.1" from @definitelytyped/dtslint-runner@0.0.134
                node_modules/@definitelytyped/dtslint-runner
                  dev @definitelytyped/dtslint-runner@"latest" from the root project
  safe-buffer@"^5.0.1" from tunnel-agent@0.6.0
  node_modules/tunnel-agent
    tunnel-agent@"^0.6.0" from request@2.88.2
    node_modules/request
      request@"^2.88.2" from @qiwi/npm-registry-client@8.9.1
      node_modules/@qiwi/npm-registry-client
        @qiwi/npm-registry-client@"^8.9.1" from @definitelytyped/utils@0.0.134
        node_modules/@definitelytyped/utils
          dev @definitelytyped/utils@"latest" from the root project
          @definitelytyped/utils@"^0.0.134" from @definitelytyped/dtslint@0.0.134
          node_modules/@definitelytyped/dtslint
            dev @definitelytyped/dtslint@"latest" from the root project
            @definitelytyped/dtslint@"^0.0.134" from @definitelytyped/dtslint-runner@0.0.134
            node_modules/@definitelytyped/dtslint-runner
              dev @definitelytyped/dtslint-runner@"latest" from the root project
          @definitelytyped/utils@"^0.0.134" from @definitelytyped/definitions-parser@0.0.134
          node_modules/@definitelytyped/definitions-parser
            dev @definitelytyped/definitions-parser@"latest" from the root project
            @definitelytyped/definitions-parser@"^0.0.134" from @definitelytyped/dtslint-runner@0.0.134
            node_modules/@definitelytyped/dtslint-runner
              dev @definitelytyped/dtslint-runner@"latest" from the root project
          @definitelytyped/utils@"^0.0.134" from @definitelytyped/dtslint-runner@0.0.134
          node_modules/@definitelytyped/dtslint-runner
            dev @definitelytyped/dtslint-runner@"latest" from the root project
  safe-buffer@"^5.2.1" from @qiwi/npm-registry-client@8.9.1
  node_modules/@qiwi/npm-registry-client
    @qiwi/npm-registry-client@"^8.9.1" from @definitelytyped/utils@0.0.134
    node_modules/@definitelytyped/utils
      dev @definitelytyped/utils@"latest" from the root project
      @definitelytyped/utils@"^0.0.134" from @definitelytyped/dtslint@0.0.134
      node_modules/@definitelytyped/dtslint
        dev @definitelytyped/dtslint@"latest" from the root project
        @definitelytyped/dtslint@"^0.0.134" from @definitelytyped/dtslint-runner@0.0.134
        node_modules/@definitelytyped/dtslint-runner
          dev @definitelytyped/dtslint-runner@"latest" from the root project
      @definitelytyped/utils@"^0.0.134" from @definitelytyped/definitions-parser@0.0.134
      node_modules/@definitelytyped/definitions-parser
        dev @definitelytyped/definitions-parser@"latest" from the root project
        @definitelytyped/definitions-parser@"^0.0.134" from @definitelytyped/dtslint-runner@0.0.134
        node_modules/@definitelytyped/dtslint-runner
          dev @definitelytyped/dtslint-runner@"latest" from the root project
      @definitelytyped/utils@"^0.0.134" from @definitelytyped/dtslint-runner@0.0.134
      node_modules/@definitelytyped/dtslint-runner
        dev @definitelytyped/dtslint-runner@"latest" from the root project

safe-buffer@5.1.2 dev optional
node_modules/are-we-there-yet/node_modules/safe-buffer
  safe-buffer@"~5.1.1" from readable-stream@2.3.7
  node_modules/are-we-there-yet/node_modules/readable-stream
    readable-stream@"^2.0.6" from are-we-there-yet@1.1.7
    node_modules/are-we-there-yet
      are-we-there-yet@"~1.1.2" from npmlog@4.1.2
      node_modules/npmlog
        optional npmlog@"2 || ^3.1.0 || ^4.0.0" from @qiwi/npm-registry-client@8.9.1
        node_modules/@qiwi/npm-registry-client
          @qiwi/npm-registry-client@"^8.9.1" from @definitelytyped/utils@0.0.134
          node_modules/@definitelytyped/utils
            dev @definitelytyped/utils@"latest" from the root project
            @definitelytyped/utils@"^0.0.134" from @definitelytyped/dtslint@0.0.134
            node_modules/@definitelytyped/dtslint
              dev @definitelytyped/dtslint@"latest" from the root project
              @definitelytyped/dtslint@"^0.0.134" from @definitelytyped/dtslint-runner@0.0.134
              node_modules/@definitelytyped/dtslint-runner
                dev @definitelytyped/dtslint-runner@"latest" from the root project
            @definitelytyped/utils@"^0.0.134" from @definitelytyped/definitions-parser@0.0.134
            node_modules/@definitelytyped/definitions-parser
              dev @definitelytyped/definitions-parser@"latest" from the root project
              @definitelytyped/definitions-parser@"^0.0.134" from @definitelytyped/dtslint-runner@0.0.134
              node_modules/@definitelytyped/dtslint-runner
                dev @definitelytyped/dtslint-runner@"latest" from the root project
            @definitelytyped/utils@"^0.0.134" from @definitelytyped/dtslint-runner@0.0.134
            node_modules/@definitelytyped/dtslint-runner
              dev @definitelytyped/dtslint-runner@"latest" from the root project
  safe-buffer@"~5.1.0" from string_decoder@1.1.1
  node_modules/are-we-there-yet/node_modules/string_decoder
    string_decoder@"~1.1.1" from readable-stream@2.3.7
    node_modules/are-we-there-yet/node_modules/readable-stream
      readable-stream@"^2.0.6" from are-we-there-yet@1.1.7
      node_modules/are-we-there-yet
        are-we-there-yet@"~1.1.2" from npmlog@4.1.2
        node_modules/npmlog
          optional npmlog@"2 || ^3.1.0 || ^4.0.0" from @qiwi/npm-registry-client@8.9.1
          node_modules/@qiwi/npm-registry-client
            @qiwi/npm-registry-client@"^8.9.1" from @definitelytyped/utils@0.0.134
            node_modules/@definitelytyped/utils
              dev @definitelytyped/utils@"latest" from the root project
              @definitelytyped/utils@"^0.0.134" from @definitelytyped/dtslint@0.0.134
              node_modules/@definitelytyped/dtslint
                dev @definitelytyped/dtslint@"latest" from the root project
                @definitelytyped/dtslint@"^0.0.134" from @definitelytyped/dtslint-runner@0.0.134
                node_modules/@definitelytyped/dtslint-runner
                  dev @definitelytyped/dtslint-runner@"latest" from the root project
              @definitelytyped/utils@"^0.0.134" from @definitelytyped/definitions-parser@0.0.134
              node_modules/@definitelytyped/definitions-parser
                dev @definitelytyped/definitions-parser@"latest" from the root project
                @definitelytyped/definitions-parser@"^0.0.134" from @definitelytyped/dtslint-runner@0.0.134
                node_modules/@definitelytyped/dtslint-runner
                  dev @definitelytyped/dtslint-runner@"latest" from the root project
              @definitelytyped/utils@"^0.0.134" from @definitelytyped/dtslint-runner@0.0.134
              node_modules/@definitelytyped/dtslint-runner
                dev @definitelytyped/dtslint-runner@"latest" from the root project
```

File: types/readable-stream/index.d.ts (hunk 1 above)
Comment (benallfree) [reply types/readable-stream/index.d.ts]: Nope just wanting to leave a breadcrumb trail in case anyone wonders about the change. I removed this

File: types/readable-stream/package.json (hunk 2 above)
Comment (jakebailey) [reply types/readable-stream/package.json:4]: Honestly I'm sort of confused why this is even here at all; safe-buffer isn't even a dependency of the package: https://github.com/nodejs/readable-stream/blob/main/package.json

Comment (benallfree) [reply types/readable-stream/package.json:4]: v2.3.7 was the last one to use it https://github.com/nodejs/readable-stream/blob/v2.3.7/package.json#L11

And I see that the typings are at v2.3 also. I guess this thing hasn't been updated in a long time. Incidentally it references safe-buffer 5.1.1 which is just a dot release behind when safe-buffer included its own types :)

Thoughts?

Comment (jakebailey) [reply types/readable-stream/package.json:4]: If the latest version doesn't use it, then I would expect the types to update to reflect that. But clearly this particular package is very out of date here too... I would maybe consider trying to fix that package as it'll fix that one too, but that is probably a big ask.

Comment (benallfree) [reply types/readable-stream/package.json:4]: @alexandercerutti started an identical PR here https://github.com/DefinitelyTyped/DefinitelyTyped/pull/62783. @alexandercerutti any interest in updating `readable-stream` typings to remove the `safe-buffer` reference and any other updates? That would be another way to fix this issue and I agree with @jakebailey that updating `readable-stream` is the ideal approach.

I'm not sure I have time to do it but if you want to jump in and help out, it would be most appreciated!

Comment (alexandercerutti) [reply types/readable-stream/package.json:4]: @benallfree I have time to handle this, but it would require me to handle things on the other PR.

> That would be another way to fix this issue and I agree with @jakebailey that updating readable-stream is the ideal approach.

You still mean DT, right?

One thing that I do not understand is why do `readable-stream` DT have `safe-buffer`? I mean, they don't have it as dependency 🤔 At least, they don't have it anymore.

Comment (alexandercerutti) [reply types/readable-stream/package.json:4]: I agree with both of you @benallfree @jakebailey. I was noticing the same thing. I'd remove safe-buffer from there. Let's solve the problem at the root! :D

I have time to handle this, but I'll have to handle on my PR. Is it okay for you, @benallfree?

Comment (benallfree) [reply types/readable-stream/package.json:4]: @alexandercerutti Agreed, let's move over to your PR and keep this one open until yours is resolved.
